tools/src/wyvern/target/corewyvernIL/support/ModuleResolver.java

```diff
@@ -359,7 +358,15 @@ private Module loadContinuation(File file, String qualifiedName, TypedAST ast, b
 
         TypeContext ctx = extendContext(Globals.getStandardTypeContext(), dependencies);
 
-        return createAdaptedModule(file, qualifiedName, valueName, dependencies, program, ctx, toplevel, loadingType);
+        boolean isModule = ast instanceof wyvern.tools.typedAST.core.declarations.ModuleDeclaration;
```

Comment:
Avoid `isModule` and the if statement by computing the isAnnotated flag earlier, when we do effect checking.  That will keep the code simpler.